private/topology/underlay/defs.go

```diff
@@ -60,7 +62,9 @@ func (o Type) String() string {
 	}
 }
 
-func TypeFromString(s string) (Type, error) {
+// @ requires low(s)
+// @ ensures  low(t) && low(err)
```

Comment:
I think that what needs to be low here is the fact that `err !=nil` and the message that may be obtained from `err`